File: app/src/main/java/dev/blackcat/minauta/async/SessionAvailableTimeAsyncTask.java
package dev.blackcat.minauta.async;

import android.content.Context;
import android.os.AsyncTask;

import dev.blackcat.minauta.data.Account;
import dev.blackcat.minauta.data.store.PreferencesStore;
import dev.blackcat.minauta.net.Connection;
import dev.blackcat.minauta.net.ConnectionFactory;
import dev.blackcat.minauta.net.JNautaConnection;

public class SessionAvailableTimeAsyncTask extends AsyncTask<Void, Void, Connection.AvailableTimeResult>
{

    public interface OnTaskResult
    {
        void onResult(Connection.AvailableTimeResult result);
    }

    Context context;
    OnTaskResult onTaskResult;

    public SessionAvailableTimeAsyncTask(Context context, OnTaskResult onTaskResult)
    {
        this.context = context;
        this.onTaskResult = onTaskResult;
    }

    @Override
    protected Connection.AvailableTimeResult doInBackground(Void... params)
    {
        PreferencesStore store = new PreferencesStore(this.context);
        Account account = store.getAccount();

        Connection connection = ConnectionFactory.createSessionProducer(JNautaConnection.class);
        Connection.AvailableTimeResult result = connection.getAvailableTime(account);

        return result;
    }

    @Override
    protected void onPostExecute(Connection.AvailableTimeResult result)
    {
        if (onTaskResult != null)
            onTaskResult.onResult(result);
    }
}

File: app/src/main/java/dev/blackcat/minauta/async/SessionAvailableTimeAsyncTask.kt
package dev.blackcat.minauta.async

import android.content.Context
import android.os.AsyncTask

import dev.blackcat.minauta.data.Account
import dev.blackcat.minauta.data.store.PreferencesStore
import dev.blackcat.minauta.net.Connection
import dev.blackcat.minauta.net.ConnectionFactory
import dev.blackcat.minauta.net.JNautaConnection

class SessionAvailableTimeAsyncTask(internal var context: Context, internal var onTaskResult: OnTaskResult?) : AsyncTask<Void, Void, Connection.AvailableTimeResult>() {

    interface OnTaskResult {
        fun onResult(result: Connection.AvailableTimeResult)
    }

    override fun doInBackground(vararg params: Void): Connection.AvailableTimeResult {
        val store = PreferencesStore(this.context)
        val account = store.account

        val connection = ConnectionFactory.createSessionProducer(JNautaConnection::class.java)

        return connection!!.getAvailableTime(account!!)
    }

    override fun onPostExecute(result: Connection.AvailableTimeResult) {
        if (onTaskResult != null)
            onTaskResult!!.onResult(result)
    }
}
